src/Interpreters/Set.cpp

```diff
@@ -278,6 +280,21 @@ void Set::checkIsCreated() const
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Trying to use set before it has been built.");
 }
 
+ColumnPtr returnFilteredColumn(const ColumnPtr & first, const ColumnPtr & second)
+{
+    ConstantFilterDescription second_const_descr(*second);
+    if (second_const_descr.always_true)
+        return second;
```

Comment:
This is not correct. The second column is a filter and has a different type. Maybe we can return nullptr in this case.